### PR TITLE
Hot store rewrite / re-set continuations

### PR DESCRIPTION
**Pull request by lukasz-golebiewski**
## Overview
See ticket

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3288

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: rspace/src/main/scala/coop/rchain/rspace/HotStore.scala [hunk 1]
```diff
@@ -39,14 +42,21 @@ final case class Cache[C, P, A, K](
...
     for {
       cached <- internalGetContinuations(channels)
       state  <- S.read
-    } yield (state.installedContinuations.get(channels) ++: cached)
+      result = state.installedContinuations.get(channels) ++: cached
+      res <- {
+        implicit val codec = fromCodec(implicitly[Codec[Seq[WaitingContinuation[P, K]]]])
+        roundTrip[F, Seq[WaitingContinuation[P, K]]](result)
+      }
+    } yield res
```
Comment (odeac) [inline rspace/src/main/scala/coop/rchain/rspace/HotStore.scala]: My understanding is that this change fixes the ignored test. Can you explain how this change affects this behaviour?

Comment (lukasz-golebiewski) [reply rspace/src/main/scala/coop/rchain/rspace/HotStore.scala]: It's described in detail in https://rchain.atlassian.net/browse/RCHAIN-3257

Comment (odeac) [reply rspace/src/main/scala/coop/rchain/rspace/HotStore.scala]: I see. The PR is now pointing to this Jira: https://rchain.atlassian.net/browse/RCHAIN-3288 which doesn't contain that information. Maybe you should link the two jiras to make it obvious for the future generations

**Comment by lukasz-golebiewski** [comment]
bors retry

**Pull request merged**